flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializationTest.java

```diff
@@ -105,11 +119,235 @@ public void testHandleMixedLargeRecords() throws Exception {
 		testSerializationRoundTrip(originalRecords, segmentSize);
 	}
 
+	/**
+	 * Non-spanning, deserialization reads one byte too many and succeeds - failure report comes
+	 * from an additional check in {@link SpillingAdaptiveSpanningRecordDeserializer}.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning1() throws Exception {
+		testHandleWrongDeserialization(
+			DeserializingTooMuch.getValue(),
+			32 * 1024);
+	}
+
+	/**
+	 * Non-spanning, deserialization reads one byte too many and fails.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning2() throws Exception {
+		testHandleWrongDeserialization(
+			DeserializingTooMuch.getValue(),
+			(serializedLength) -> serializedLength,
+			isA(IndexOutOfBoundsException.class));
```

Comment:
Why are you codifying a contract that sometimes `IndexOutOfBoundsException` is thrown and sometimes `EOFException`?

Reply:
This is the current behaviour (non-spanning accesses arrays and spanning provides an input view that resembles file/stream accesses, each with the respective exceptions in case of failures)...
Do you think we should should be more generic and tolerate any `IOException` here? But then the cause may not be right...

Reply:
Did you mean `Exception` instead of `IOException`? I think that we should either clean this up and throw only one type of an exception (`DeserialisationException`?) or if we are throwing "unspecified" exceptions, we should test just for that (`isA(Exception.class)`).

Again I would like to avoid the scenario when someone in the future makes a good refactor/clean up, one of those tests starts to fail, and our future programmer will have to spend time thinking whether throwing exactly `IndexOutOfBoundsException` in one case and `EOFException` in other was part of some important contract. This is what I meant by "codifying a contract". Such kind of tests might cause someone to tip-toe/triple think "what is happening here?".